server/grpc_test.go

```diff
@@ -41,6 +42,7 @@ var (
 	casClient pb.ContentAddressableStorageClient
 	bsClient  bytestream.ByteStreamClient
 	ctx       = context.Background()
+	diskCache *disk.DiskCache
```

Comment:
Would this cause trouble if we were to run tests concurrently?

Reply:
It seems to me the disk cache is safe for concurrent use

Reply:
The disk cache is safe for concurrent use, but I think it's possible for two concurrent tests to confuse each other if they share a cache (stats will be different, items can be evicted etc). Instead of doing `diskCache.Get()` the new test could do a bytestream `Read()` call, and then we don't need a package level `diskCache` variable.

Reply:
Would it not still be the case that the disk cache is still shared across tests through the single grpc server instance?

It looks like we would have to move the entirety of TestMain into a setup function which is called before each individual test.

Reply:
Hmm, you're right. I don't use TestMain often, and for some reason I thought that it was run once per test. Sorry about the distraction.